lib/cocoapods/installer/xcode/pods_project_generator.rb

```diff
@@ -258,6 +259,23 @@ def configure_app_extension_api_only_for_target(target)
             config.build_settings['APPLICATION_EXTENSION_API_ONLY'] = 'YES'
           end
         end
+
+        def user_swift_version
+          declared_swift_versions = analysis_result
+            .targets
+            .map(&:user_project)
+            .compact
+            .flat_map(&:native_targets)
+            .flat_map(&:build_configurations)
+            .flat_map(&:build_settings)
+            .map { |settings| settings['SWIFT_VERSION'] }
+            .compact
+            .uniq
+
+          # TODO: Assert only 1 or 0 unique Swift Versions declared
+
+          declared_swift_versions.first
```

Comment:
@orta @DanToml Just wanted to point out that, according to [Apple's blog post on the matter](https://developer.apple.com/swift/blog/?id=36), an app cannot use Swift 2.3 and 3.0 together, due to the binary incompatibility. `First, Swift 2.3 and Swift 3 are not binary compatible so your app's entire code base needs to pick one version of Swift.` So I'm thinking this can really just be a check of whatever version of Swift the project is using can be set on all of the pods.